tests/data_context/test_get_data_context.py

```diff
@@ -0,0 +1,230 @@
+import os
+import pathlib
+from unittest import mock
+
+import pytest
+
+import great_expectations as gx
+from great_expectations import DataContext
+from great_expectations.data_context import BaseDataContext, CloudDataContext
+from great_expectations.data_context.types.base import DataContextConfig
+from great_expectations.exceptions import ConfigNotFoundError, DataContextError
+
+
+@pytest.mark.unit
+def test_config_returns_base_context():
+    config: DataContextConfig = DataContextConfig(
+        config_version=3.0,
+        plugins_directory=None,
+        evaluation_parameter_store_name="evaluation_parameter_store",
+        expectations_store_name="expectations_store",
+        datasources={},
+        stores={
+            "expectations_store": {"class_name": "ExpectationsStore"},
+            "evaluation_parameter_store": {"class_name": "EvaluationParameterStore"},
+            "validation_result_store": {"class_name": "ValidationsStore"},
+        },
+        validations_store_name="validation_result_store",
+        data_docs_sites={},
+        validation_operators={},
+    )
+    assert isinstance(gx.get_context(project_config=config), BaseDataContext)
+
+
+@pytest.mark.unit
+def test_empty_call_returns_data_context(tmp_path: pathlib.Path):
+    project_path = tmp_path / "empty_data_context"
+    project_path.mkdir()
+    project_path_str = str(project_path)
+    gx.data_context.DataContext.create(project_path_str)
+    os.chdir(project_path_str)
```

Comment:
How does the working dir get changed back?

If we don't already have this, I would recommend a context manager utility method that automatically resets the working directory when it exits.

https://stackoverflow.com/a/431747/6304433
```python
import pathlib
import os
from contextlib import contextmanager

@contextmanager
def working_directory(directory: str | pathlib.Path):
    owd = os.getcwd()
    try:
        os.chdir(directory)
        yield directory
    finally:
        os.chdir(owd)
```

Reply:
Love it. Added to `test_utils.py` and updated current test